### PR TITLE
feat(instrumentation-pg): add error type to db duration metric

### PR DESCRIPTION
**Pull request by maryliag**
## Which problem is this PR solving?

- Part Of #2348 

## Short description of the changes

- Add error type to db duration operation metric. It uses the error codes from postgres, so we can limit cardinality and they're well known error codes.

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-pg/src/utils.ts [hunk 1]
```diff
@@ -252,6 +255,10 @@ export function patchCallback(
     res: object
   ) {
     if (err) {
+      if (Object.keys(err).includes('code')) {
```
Comment (david-luna) [inline plugins/node/opentelemetry-instrumentation-pg/src/utils.ts:258]: nit: maybe is not necessary to create an array of keys to check the presence of `code`. I think you could make use of `Object.hasOwnProperty`

File: plugins/node/opentelemetry-instrumentation-pg/test/pg.test.ts [hunk 2]
```diff
@@ -1013,5 +1014,53 @@ describe('pg', () => {
...
+        assert.strictEqual(
+          metrics[0].descriptor.description,
+          'Duration of database client operations.'
+        );
+        const dataPoint = metrics[0].dataPoints[0];
+        assert.strictEqual(
+          dataPoint.attributes[SEMATTRS_DB_SYSTEM],
+          DBSYSTEMVALUES_POSTGRESQL
+        );
+        // TODO: add once #2475 is merged
```
Comment (david-luna) [inline plugins/node/opentelemetry-instrumentation-pg/test/pg.test.ts:1042]: It's approved :)

**Review by david-luna — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-pg/test/pg.test.ts (hunk 2 above)
Comment (pichlermarc) [reply plugins/node/opentelemetry-instrumentation-pg/test/pg.test.ts:1042]: https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2475 is now merged too :slightly_smiling_face:

Comment (maryliag) [reply plugins/node/opentelemetry-instrumentation-pg/test/pg.test.ts]: Merged with main and added the code

**Pull request merged**